app/controllers/api/proposals_controller.rb

```diff
@@ -0,0 +1,109 @@
+module Api
+  class ProposalsController < BaseController
+    rescue_from ActiveRecord::RecordNotFound, with: :proposal_not_found_error
+    rescue_from ClientModelMismatch, with: :auth_errors
+
+    before_action -> { authorize proposal }, only: [:update, :show]
+
+    def create
+      if proposal = create_proposal
+        DispatchFinder.run(proposal).deliver_new_proposal_emails
+        render json: ProposalSerializer.new(proposal).to_json
+      else
+        render json: {
+          errors: errors,
+          message: "Failed to create proposal"
+        }, status: 400
+      end
+    end
+
+    def update
+      if update_proposal
+        render json: ProposalSerializer.new(proposal).to_json
+      else
+        render json: {
+          errors: errors,
+          message: "Failed to update proposal"
+        }, status: 400
+      end
+    end
+
+    def show
+      render json: ProposalSerializer.new(proposal).to_json
+    end
+
+    def index
+      listing = ProposalListingQuery.new(current_user, params)
+      begin
+        proposals = listing.query
+        render json: ProposalSearchSerializer.new(proposals).to_json
+      rescue SearchBadQuery, SearchUnavailable => error
+        render json: { error: error.message }, status: 500
```

Comment:
should we include a `message` key  with a generic value here as well to be consistent with other error json in this controller?